### PR TITLE
OSSM-3731: Add api postsubmit jobs

### PR DESCRIPTION
**Pull request by leoieggli**

### CODE REVIEW

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-api-maistra-2.2-api-update-istio-2-2

**Comment by yxun** [comment]
/pj-rehearse pull-ci-maistra-api-maistra-2.2-api-update-istio-2-2

**Comment by yxun** [comment]
I used a workaround to verify the job by copying the current latest api maistra-2.2 branch commit SHA and assigning it to the temporary var.
After review, I will remove the workaround in another commit for the PR.

File: ci-operator/config/maistra/api/maistra-api-maistra-2.4.yaml [hunk 1]
```diff
@@ -30,16 +30,24 @@ tests:
...
-    ./maistra/run-test-infra-script.sh ./tools/automator.sh \
-    -o maistra \
-    -f /creds-github/github-token \
-    -r istio \
-    -c UPDATE_BRANCH=$AUTOMATOR_SHA bin/update_maistra_deps.sh \
-    '-t Automator: Update maistra/api dependency' \
-    -m update_api_istio \
-    -b maistra-2.4
+    export BUILD_WITH_CONTAINER="0"
+    export GOFLAGS="-mod=vendor"
```
Comment (jwendell) [inline ci-operator/config/maistra/api/maistra-api-maistra-2.4.yaml:36]: remove this. 2.4 doesn't use vendor

**Review by jwendell — CHANGES_REQUESTED** [review]
LGTM.

Remove the workaround, add the postsubmit: true and remove an env variable from 2.4.

**Review by jwendell — APPROVED** [review]

**Pull request merged**